DaisyMicheal/Ecommerce_web
# Implement full login functionality with API integration

**Pull Request Description for Login Page**

**What does this PR do?**
Create and style the login page for the e-commerce platform.

**Description of Task to be completed:**
- Design and implement the login page layout, including fields for email and password.
- Integrate form validation to ensure users provide valid login credentials.
- Implement error handling and display user-friendly error messages for incorrect credentials.
- Add a "Login" button that authenticates the user and redirects them to the appropriate page upon successful login.
- Include options for users to log in using social media accounts (e.g., Google, Facebook).
- Implement a "Forgot Password" link to allow users to reset their password.
- Ensure the login page is fully responsive and accessible across different devices.
- Write unit tests to cover the form validation, authentication process, and error handling.
- Securely handle user credentials and integrate with the backend for user authentication.

**How should this be manually tested?**
- Pull the branch and install dependencies if necessary.
- Navigate to the login page and enter valid and invalid credentials to test the form validation and authentication process.
- Attempt to log in using both form fields and social media options.
- Verify that error messages are displayed for incorrect credentials and that the user is redirected correctly upon successful login.
- Test the "Forgot Password" functionality to ensure it redirects to the appropriate page.
- Run the unit tests to ensure they pass.

**Any background context you want to provide?**
None

**What are the relevant Pivotal Tracker stories?**
#188043212

**Screenshots (if appropriate):**
[image]

**Questions:**
None
